src/Artemis.UI/Screens/Debugger/Tabs/Logs/LogsDebugView.axaml.cs

```diff
@@ -1,17 +1,72 @@
+using System;
+using System.Diagnostics;
+using System.Reflection;
+using Avalonia;
+using Avalonia.Controls;
+using Avalonia.Controls.Primitives;
 using Avalonia.Markup.Xaml;
 using Avalonia.ReactiveUI;
+using AvaloniaEdit;
 
 namespace Artemis.UI.Screens.Debugger.Logs;
 
 public class LogsDebugView : ReactiveUserControl<LogsDebugViewModel>
 {
+    private int _lineCount;
+    private TextEditor _textEditor;
+
     public LogsDebugView()
     {
+        _lineCount = 0;
         InitializeComponent();
     }
 
     private void InitializeComponent()
     {
         AvaloniaXamlLoader.Load(this);
+        _textEditor = this.FindControl<TextEditor>("log");
+    }
+
+    protected override void OnAttachedToVisualTree(VisualTreeAttachmentEventArgs e)
+    {
+        base.OnAttachedToVisualTree(e);
+        _textEditor?.ScrollToEnd();
+    }
+
+    private void OnTextChanged(object? sender, EventArgs e)
+    {
+        if (_textEditor is null)
+            return;
+        if (_textEditor.ExtentHeight == 0)
+            return;
+
+        var linesAdded = _textEditor.LineCount - _lineCount;
+        var lineHeight = _textEditor.ExtentHeight / _textEditor.LineCount;
+        var outOfScreenTextHeight = _textEditor.ExtentHeight - _textEditor.VerticalOffset - _textEditor.ViewportHeight;
+        var outOfScreenLines = outOfScreenTextHeight / lineHeight;
```

Comment:
Just a bit of code style
```suggestion
        int linesAdded = _textEditor.LineCount - _lineCount;
        double lineHeight = _textEditor.ExtentHeight / _textEditor.LineCount;
        double outOfScreenTextHeight = _textEditor.ExtentHeight - _textEditor.VerticalOffset - _textEditor.ViewportHeight;
        double outOfScreenLines = outOfScreenTextHeight / lineHeight;
```